### DISCUSSION TITLE
Current assessment of the road map to Scala 3

### DISCUSSION THREAD

**Discussion by som-snytt** [scala/scala3#19194] [General Discussion]
The blog about to what to expect from the transition to Scala 3:

https://scala-lang.org/2019/12/18/road-to-scala-3.html

What is the current assessment or understanding about migration with respect to the rubrics of that document?

Or perhaps there is a new set of concerns that has emerged since 2019.

One may omit, "What about performance?"

**Comment by SethTisue** [comment]
Are there particular items that concern you? I skimmed through it and nothing obvious jumped out at me at having been missed.

SIP-51 (on dropping forward compat of Scala 2 stdlib)  is still pending, of course, is an obvious area of concern, but it's expected to move forward.

**Comment by som-snytt** [comment]
The document is a fait accompli, 3.0 has been released. But it only takes us to 3.0. Now item 2 is about the library. 2.14 was predicated, at the time in this document, on whether the library evolves. What is the current outlook? Maybe there is nothing more to say except to reference the SIP. What about tooling, as we hear from the gallery? (Item 8.) I'm no longer worried about Python 2/3 syndrome (item 9), as we have the long covid form of migration. It will affect everyone's daily activities for some time. But it seems appropriate to take a moment before the New Year to summarize how the parts fit together for the middle part of the decade. It's also an opportunity to rehearse the successes of the previous road map.

**Comment by som-snytt** [comment]
It's not just me:

Now, the general vibe of Scala seems to be more of the Haskell one with an addition of "we don't know what's happening and where we're going, although, for some people it's real fun 🤪".

https://discord.com/channels/632150470000902164/632628675287973908/1184801768861880381

I don't (necessarily) agree with that assessment; I just hope the invisible hand finds time to document the vision. (Maybe there is already a Scala Days keynote somewhere.)
